src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx

```diff
@@ -38,12 +39,25 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
 
   const [newFilterName, setNewFilterName] = useState(name);
   const [isEditing, setIsEditing] = useState(false);
+  const [isValid, setIsValid] = useState(true);
 
   const resetEditState = () => {
     setIsEditing(false);
     setNewFilterName(name);
   };
 
+  const handleSubmit = () => {
+    if (isValid) {
+      editFilter("name", newFilterName, filter);
+      resetEditState();
+    }
+  };
+
+  const validationMessage =
+    newFilterName === ""
+      ? "Filter cannot be empty"
+      : "Invalid Regular Expression";
```

Comment:
nit. assuming the error messages should follow the same capitalization scheme
```suggestion
      : "Invalid regular expression";
```